src/logging.cpp

```diff
@@ -121,7 +144,12 @@ bool BCLog::Logger::DisableCategory(const std::string& str)
 
 bool BCLog::Logger::WillLogCategory(BCLog::LogFlags category) const
 {
-    return (m_categories.load(std::memory_order_relaxed) & category) != 0;
+    Assume(category < NONE);
+    if (category < NONE) {
```

Comment:
```suggestion
   
    if ( Assume(category < NONE)) {
```

nit: Can be combined, IIRC